sentry/src/main/java/io/sentry/cache/CacheStrategy.java

```diff
@@ -120,13 +120,13 @@ private void moveInitFlagIfNecessary(
       final @NotNull File currentFile, final @NotNull File[] notDeletedFiles) {
     final SentryEnvelope currentEnvelope = readEnvelope(currentFile);
 
-    if (!isValidEnvelope(currentEnvelope)) {
+    if (currentEnvelope != null && !isValidEnvelope(currentEnvelope)) {
```

Comment:
`isValidEnvelope` already does NPE check

Reply:
I believe the static analyzer can't follow the flow of null check through the private method calls so the check is needed to make the compiler happy. aka: tool limitation